tests/clientlib_test.py

```diff
@@ -166,6 +166,54 @@ def test_validate_warn_on_unknown_keys_at_top_level(tmpdir, caplog):
     ]
 
 
+@pytest.mark.parametrize(
+    ('rev', 'error'),
+    (
+        ('HEAD', True),
+        ('stable', True),
+        ('master', True),
+        ('v0.12.4', False),
+        ('b27f281eb9398fc8504415d7fbdabf119ea8c5e1', False),
+        ('19.10b0', False),
+        ('4.3.21-2', False),
+    ),
+)
+def test_check_mutable_revs(caplog, rev, error):
+    config_obj = {
+        'repo': 'https://gitlab.com/pycqa/flake8',
+        'rev': rev,
+        'hooks': [{
+            'id': 'flake8',
+        }],
+    }
+    cfgv.validate(config_obj, CONFIG_REPO_DICT)
+
+    expected = [
+        (
+            'pre_commit',
+            logging.WARNING,
+            "The 'rev' field of repo 'https://gitlab.com/pycqa/flake8' "
+            "shouldn't be mutable. See https://pre-commit.com/#using-the-"
+            'latest-version-for-a-repository for further details.',
+        ),
+    ] if error else []
```

Comment:
please don't put logic in tests, this should be two separate tests, one demonstrating positive cases and one demonstrating negatives -- as it is right now you've introduced a boolean trap via parametrize (a common antipattern)